### PR TITLE
sub/sd_lavc: check decoder output type for dvb and arib

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: sub/sd_lavc.c [hunk 1]
```diff
@@ -99,6 +100,29 @@ static int init(struct sd *sd)
...
+        int ret = av_opt_get_int(ctx, "sub_type", AV_OPT_SEARCH_CHILDREN, &format);
+        if (!ret && format != SUBTITLE_BITMAP) {
+            avcodec_free_context(&ctx);
+            return -1;
+        }
+        break;
+    }
+    }
+
+    priv = talloc_zero(NULL, struct sd_lavc_priv);
```
Comment (sfan5) [inline sub/sd_lavc.c:132]: move the declaration too?

Comment (kasper93) [reply sub/sd_lavc.c:132]: There is goto inbetween to error, which uses priv

File: sub/lavc_conv.c [hunk 2]
```diff
@@ -82,14 +81,30 @@ struct lavc_conv *lavc_conv_create(struct mp_log *log,
...
         goto error;
 
     priv->avpkt = av_packet_alloc();
     priv->avpkt_vtt = av_packet_alloc();
     if (!priv->avpkt || !priv->avpkt_vtt)
         goto error;
 
+    switch (codec->id) {
+    case AV_CODEC_ID_DVB_TELETEXT:;
+        if (!sd->opts->teletext_page) {
```
Comment (kasper93) [inline sub/lavc_conv.c:94]: I changed my mind, will move this to decode(). Just so we can update page number during runtime. I think there is value in that.

Comment (kasper93) [reply sub/lavc_conv.c]: done

File: sub/lavc_conv.c [hunk 3]
```diff
@@ -82,14 +83,23 @@ struct lavc_conv *lavc_conv_create(struct mp_log *log,
...
+    if (mp_set_avctx_codec_headers(avctx, sd->codec) < 0)
         goto error;
 
     priv->avpkt = av_packet_alloc();
     priv->avpkt_vtt = av_packet_alloc();
     if (!priv->avpkt || !priv->avpkt_vtt)
         goto error;
 
+    switch (codec->id) {
+    case AV_CODEC_ID_DVB_TELETEXT:;
```
Comment (sfan5) [inline sub/lavc_conv.c:95]: stray `;`

**Review by sfan5 — APPROVED** [review]
didn't look closely but seems fine

File: sub/lavc_conv.c (hunk 3 above)
Comment (kasper93) [reply sub/lavc_conv.c]: good catch, no longer needed after moving the code.

**Pull request merged**